### PR TITLE
test(init): initialize vitest for client and shared

### PR DESCRIPTION
**Pull request by Nithos**

### CODE REVIEW

File: packages/client/tests/unit/__snapshots__/basic.spec.ts.snap [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Vitest Snapshot v1
```
Comment (YohannParis) [inline packages/client/tests/unit/__snapshots__/basic.spec.ts.snap:1]: Is this suppose to be on git? it is an automatically generated file?

**Review by YohannParis — APPROVED** [review]

File: packages/client/tests/unit/__snapshots__/basic.spec.ts.snap (hunk 1 above)
Comment (Nithos) [reply packages/client/tests/unit/__snapshots__/basic.spec.ts.snap:1]: the basic tests does a dummy snapshot that is generated yes, now that its there any changes to the code/test will compare to that snapshot. if it does not match it will fail the test (so you would need to update the snapshot if its a valid change)

TLDR; yes it is generated but it should be there as it is used for test comparison
NOTE: this is a dummy test so has no bearing on the app so can be removed, was meant to showcase the usage at this point

**Review by YohannParis — APPROVED** [review]
